src/propstepsolvers.jl

```diff
@@ -34,34 +34,36 @@ QuKrylov() = QuKrylov(@compat Dict(:NC=>10))
 Propagates to the next time state
 Input Parameters:
 `prob`             :  Method to be used
-`op`               :  Hamiltonian of the system
+`eq`               :  Quantum Equation type
 `t`                :  Time corresponding to the t_state
 `current_t`        :  Current time
 `current_qustate`  :  Quantum state corresponding to current time
 """ ->
-function propagate(prob::QuEuler, op, t, current_t, current_qustate)
+function propagate(prob::QuEuler, eq::QuEquation, t, current_t, current_qustate)
     dt = t - current_t
-    return (eye(op)-im*op*dt)*current_qustate
+    return (eye(operator(eq))-im*operator(eq)*dt)*vec(current_qustate)
```

Comment:
It is better to do `op = operator(eq)` before this statement and then only use `op`. This way you only call `operator` once.